src/node_file.cc

```diff
@@ -1846,12 +1846,12 @@ static void Read(const FunctionCallbackInfo<Value>& args) {
   char* buffer_data = Buffer::Data(buffer_obj);
   size_t buffer_length = Buffer::Length(buffer_obj);
 
-  CHECK(args[2]->IsInt32());
-  const size_t off = static_cast<size_t>(args[2].As<Int32>()->Value());
+  CHECK(args[2]->IsNumber());
```

Comment:
Saying it's an int64 and using `IsNumber()` is wrong-ish because the latter signals it's a double and those are only precise up to 2**53-1.

I wouldn't be surprised if we conflate them in other places too. Perhaps it's time for V8 to grow an `IsInt64()` method or for us to grow a utility method that checks the same.

Reply:
It's already checked in the JS wrapper; is that not enough?

(I copied what's done for `position` a few lines down.)

Reply:
> It's already checked in the JS wrapper; is that not enough?

It's fragile and non-obvious, IMO.

Aside: maybe `validateInteger()` should be renamed to `validateSafeInteger()` - just 'integer' doesn't convey much except that it rejects rationals.

> It's also tested on the next line to be less than the buffer length (guaranteed to be safe int).

That's another buglet: `size_t` might be 32 bits. Narrowing from `int64_t` to `size_t` can be lossy.

Reply:
Is this close to what you have in mind?
```cpp
void CheckSafePositiveJsInt(v8::Value v) {
  CHECK(v->IsNumber());
  double v_d = v.As<Number>()->Value();
  CHECK_EQ(std::trunc(v_d), v_d); // is int
  int64_t v_i64 = static_cast<int64_t>(v_d);
  CHECK_GE(v_i64, 0); // is positive int
  CHECK_LE(v_i64, MAX_SAFE_JS_INTEGER); // is positive safe JS int
}

// in Read()
CheckSafePositiveJsInt(args[2]);
// now safe to cast to uint64_t (likely equivalent of size_t)
uint64_t off_u64 = static_cast<uint64_t>(args[2].As<Integer>()->Value());
CHECK_LT(off_u64, buffer_length); // buffer_length promoted on 32-bit arch
// buffer_length is size_t, so now safe to cast to size_t
size_t off = static_cast<size_t>(off_u64);

```

Reply:
Yes, that looks good.